UniWatch/DataAccess/ClassManager.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UniWatch.Models;
+using System.Data.Entity;
+
+namespace UniWatch.DataAccess
+{
+    /// <summary>
+    /// Provides central access to class related information
+    /// </summary>
+    public class ClassManager : IClassManager
+    {
+        private bool disposed = false;
+        private AppDbContext _db;
+
+        /// <summary>
+        /// Creates a default ClassManager
+        /// </summary>
+        public ClassManager()
+        {
+            _db = new AppDbContext();
+        }
+
+        /// <summary>
+        /// Creates a ClassManager with the given context
+        /// </summary>
+        /// <param name="context">The context to create the manager with</param>
+        public ClassManager(AppDbContext context)
+        {
+            _db = context;
+        }
+
+        /// <summary>
+        /// Creates a new class
+        /// </summary>
+        /// <param name="name">The name/title of the class</param>
+        /// <param name="number">The course number</param>
+        /// <param name="section">The section</param>
+        /// <param name="semester">The semester e.g. Fall 2016</param>
+        /// <param name="teacher">The teacher for this class</param>
+        /// <returns>The created class</returns>
+        /// <remarks>Throws InvalidOperationException if class already exists with the given details</remarks>
+        public Class CreateClass(string name, int number, string section, Semester semester, int year, int teacherId)
+        {
+            var existing = _db.Classes
```

Comment:
Note, you can put everything in `.Count()`. Nothing is wrong, just a different way of slicing the cake.

```
_db.Classes
    .Count(c => c.Name == name &&
        c.Number == number &&
        c.Section == section &&
        c.Semester == semester &&
        c.Teacher.Id == teacherId);
```